entgql/extension.go

```diff
@@ -283,63 +277,55 @@ func (e *Extension) hasMapping(f *gen.Field) (string, bool) {
 	}
 }
 
-// isInput reports if the given type is an input object.
-func (e *Extension) isInput(name string) bool {
-	if t, ok := e.cfg.Schema.Types[name]; ok && t != nil {
-		return t.IsInputType()
-	}
-	return false
-}
-
 // genSchema returns a new hook for generating
 // the GraphQL schema from the graph.
-func (e *Extension) genSchema() gen.Hook {
+func (e *Extension) genSchemaHook() gen.Hook {
 	return func(next gen.Generator) gen.Generator {
 		return gen.GenerateFunc(func(g *gen.Graph) error {
 			if err := next.Generate(g); err != nil {
 				return err
 			}
 
+			if !(e.genSchema || e.genWhereInput) {
+				return nil
+			}
```

Comment:
```suggestion
			if !e.genSchema && !e.genWhereInput {
				return nil
			}
```
Feels more intuitive to me.